### ISSUE TITLE
[$250] Expense- RBR gets shown for advanced approval enabled expense if default currency is changed

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#48328]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.26-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
Issue was found when executing this PR: https://github.com/Expensify/App/pull/47926
**Email or phone of affected tester (no customers):** Owner: [applausetester+nl518@applause.expensifail.com](mailto:applausetester+nl518@applause.expensifail.com)
Submitter: [applausetester+nl520@applause.expensifail.com](mailto:applausetester+nl520@applause.expensifail.com)
**Issue reported by:** Applause Internal Team

## Action Performed:
Pre-condition: Default workspace currency should be other currency than USD.

1. Navigate to Old Dot (staging.expensify.com/inbox)
2. Sign up with a new account
3. Create a control workspace
4. Enable workspace chat for new dot by running the code below inside the console.
p = Policy.getCurrent();
p.policy.isPolicyExpenseChatEnabled = "true";
p.save();
5. Go to reports and change the default currency to USD
6. Go to members and enable advanced approvals
7. Invite a member as an employee and make it submit to the owner
8. Invite another member as an employee but make it submit to the member invited on step 7
9. Navigate to New Dot (staging.new.expensify.com)
10. Sign in as the member (submitter) invited on step 8.
11. Go to workspace chat and submit a manual expense

## Expected Result:
The submitted expense doesn't show any errors or RBR

## Actual Result:
RBR is shown on the expense preview component, on the LHN and there is an error inside the expense report.

## Workaround:
Unknown

## Platforms:

- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/bf803f2c-b8e0-441c-baab-7f45227acc79

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01510356bd8e22c968</li>
        <li>Upwork Job ID: 1829578766264533133</li>
        <li>Last Price Increase: 2024-09-06</li>
<li>Automatic offers: </li>
<ul>
<li>akinwale | Reviewer | 103886034</li>
</ul></li>
    </ul>
</details>

**Comment by IuliiaHerets** [comment]
We think that this bug might be related to #wave-control

**Comment by IuliiaHerets** [comment]
@sakluger  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by sakluger** [comment]
I'm adding the external label, but I wonder if this is a BE error.

**Comment by sakluger** [comment]
No proposals yet.

**Comment by sakluger** [comment]
Still no proposals. I'll post in wave-control in Slack on Monday if we don't get any by then.

**Comment by kubabutkiewicz** [comment]
Hi, I’m Jakub from Callstack and I would like to work on this issue.

**Comment by sakluger** [comment]
Thanks @kubabutkiewicz! Let us know what you find.

**Comment by kubabutkiewicz** [comment]
Hey, I am not able to reproduce original issue. @IuliiaHerets are you still able to reproduce that? 
But while investigating I stumble upon two other issues, one is I think backend issue.  
I went through steps to reproduce original issue after submitting money request it got automatically approved, while every next request didn't. When I checked response from `OpenReport` command it had message that request is approved
 
<img width="1885" alt="SCR-20240910-labb" src="https://github.com/user-attachments/assets/5b2adc9f-3bcd-4e8b-9470-b3dc012b7d02">

Second seems to be a frontend issue, after creating a money request it the preview has RBR but when I go to the details I don't see any message why it have RBR, after checking code and onyx state it seems to be caused by this 
<img width="1892" alt="SCR-20240910-knpz" src="https://github.com/user-attachments/assets/cae97303-d209-4850-bfd4-6d24d3bd79df">
And this is added here as an optimistic update https://github.com/Expensify/App/blob/fe4d19e592d68684fb471f2bba93787c88a3a3f3/src/libs/actions/IOU.ts#L472-L502
But I believe its different than the original issue and this is not reproducible constantly 
Should I work on the second issue then? @sakluger
